From d9532e1d665293441d8806a60abedbad990dad80 Mon Sep 17 00:00:00 2001
From: isabel <isabel@isabelroses.com>
Date: Fri, 9 May 2025 10:47:49 +0100
Subject: [PATCH] fix: remove lix required features

---
 src/checks.rs | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/checks.rs b/src/checks.rs
index bf805f0..1178f20 100644
--- a/src/checks.rs
+++ b/src/checks.rs
@@ -73,11 +73,6 @@ pub fn check_nix_features() -> Result<()> {
 
     let mut required_features = vec!["nix-command", "flakes"];
 
-    // Lix still uses repl-flake, which is removed in the latest version of Nix.
-    if util::is_lix()? {
-        required_features.push("repl-flake");
-    }
-
     tracing::debug!("Required Nix features: {}", required_features.join(", "));
 
     // Get currently enabled features
-- 
2.49.0

